src/wolfio.c

```diff
@@ -1486,6 +1486,30 @@ WOLFSSL_API void wolfSSL_CTX_SetIOSend(WOLFSSL_CTX *ctx, CallbackIOSend CBIOSend
 }
 
 
+/* sets the IO callback to use for receives at WOLFSSL level */
+WOLFSSL_API void wolfSSL_SSLSetIORecv(WOLFSSL *ssl, CallbackIORecv CBIORecv)
```

Comment:
Why do we now need this at the SSL level?  How is priority set between CTX and SSL?  Is handled perfectly in all case?  Where is that documented?

Reply:
It makes it easier for the case of one JSSE WOLFSSL session being created using sockets and one being created using the "engine" both from the same WOLFSSL_CTX. Both have their separate different callbacks to be used but are created from the same context.

Setting at the WOLFSSL level overwrites the CB pointer passed down from the CTX during InitSSL(). 

It can be solved at the JSSE implementation level if wanted.

Reply:
Ok, looks like only 1 call to each callback and they're through the ssl-> pointer.